File: Samsung_SW_Expert_Academy/D2/2007. 패턴 마디의 길이.cpp
#include<iostream>
#include<string>
using namespace std;

int main(int argc, char** argv)
{
	int test_case;
	int T;
	cin>>T;
	for(test_case = 1; test_case <= T; ++test_case)
	{
		string s;
        cin >> s;
        int L=2;
        while(1){
            bool check=true;
         
            for(int i=0; i<30-L; i++){
                if(s[i]!=s[i+L]){
                    check=false;
                    break;
                }
            }
            if(check){
                cout <<"#"<<test_case<<" "<<L<<"\n";
                break;
            }
            L++;
        }
	}
	return 0;
}

File: Samsung_SW_Expert_Academy/D2/2007. 패턴 마디의 길이.java

import java.util.Scanner;
import java.io.FileInputStream;

class Solution {
    public static void main(String args[]) throws Exception {
        Scanner sc = new Scanner(System.in);
        int T;
        T = sc.nextInt();
        for (int test_case = 1; test_case <= T; test_case++) {
            String s = sc.next();
            int L = 2;
            while (true) {
                Boolean check = true;
                for (int i = 0; i < 30 - L; i++) {
                    if (s.charAt(i) != s.charAt(i + L)) {
                        check = false;
                        break;
                    }
                }
                if (check) {
                    System.out.println("#" + test_case + " " + L);
                    break;
                } else
                    L++;
            }

        }
    }
}